graylog2-web-interface/src/components/indices/IndexSetFieldTypes/IndexSetCustomFieldTypeRemoveModal.tsx

```diff
@@ -68,11 +71,51 @@ const indexSetsStoreMapper = ({ indexSets }: IndexSetsStoreState): Record<string
 const IndexSetCustomFieldTypeRemoveContent = ({ fields, indexSets, setRotated, rotated, indexSetIds }: ContentProps) => {
   const fieldsStr = fields.join(', ');
   const indexSetsStr = indexSetIds.map((id) => indexSets[id].title).join(', ');
+  const { customFieldMappingsByField, name: profileName, id: profileId } = useIndexProfileWithMappingsByField();
+  const { overriddenIndexFieldsStr, overriddenProfilesFieldsWithType } = useMemo<{ overriddenIndexFieldsStr: string, overriddenProfilesFieldsWithType: Array<{ field: string, type: string }> }>(() => {
+    const { overriddenIndexFields, overriddenProfilesFields } = fields.reduce((acc, cur) => {
+      if (customFieldMappingsByField[cur]) {
+        acc.overriddenProfilesFields.push({ field: cur, type: customFieldMappingsByField[cur] });
+      } else {
+        acc.overriddenIndexFields.push(cur);
+      }
+
+      return acc;
+    }, { overriddenIndexFields: [], overriddenProfilesFields: [] });
+
+    return ({
+      overriddenIndexFieldsStr: overriddenIndexFields.join(', '),
+      overriddenProfilesFieldsWithType: overriddenProfilesFields,
+    });
+  }, [customFieldMappingsByField, fields]);
```

Comment:
It took me a moment to understand the reduce block. Maybe we can make it easier to read by doing something like:

```
const overriddenProfilesFieldsWithType = // iterate over customFieldMappingsByField
const overriddenIndexFieldsStr = // iterate over fields and do not add fields which are part of customFieldMappingsByField
```

We would have another iteration compared to one, but it would still be very performant, the code path would be linear and we can get rid of reduce and do not have to create a new object for every iteration.

Reply:
@linuspahl `do not have to create a new object for every iteration.` - Are you talking about `{ field: cur, type: customFieldMappingsByField[cur] }` or reduce accumulator? 
The first one we would have to create in any case, because `customFieldMappingsByField`  is an object `<Field, Type>`.  I can try to iterate by `data.customFieldMappings` from `useProfile` 
In the second case, it would be the same object on each iteration.

But in the end, I can make it more readable by adding a helper function and some extra var like `isInProfile`

`
/// in component 
const { overriddenIndexFieldsStr, overriddenProfilesFieldsWithType } = useMemo(separateByOrigin(fields, customFieldMappingsByField), [ fields, customFieldMappingsByField ])
//// in helpers 

const separateByOrigin = (fields, customFieldMappingsByField) => {
return fields.reduce((acc, cur) => {
      const isInProfile = !!customFieldMappingsByField[cur])

      if (isInProfile) {
        acc.overriddenProfilesFields.push({ field: cur, type: customFieldMappingsByField[cur] });
      } else {
        acc.overriddenIndexFields.push(cur);
      }

      return acc;
    }
}
`